regnety/config/config.py

```diff
@@ -0,0 +1,116 @@
+from dataclasses import dataclass
+from typing import List
+
+ALLOWED_FLOPS = ('200mf', '400mf', '600mf', '800mf')
+
+@dataclass
+class RegNetYConfig:
+    """
+    Dataclass for architecture configuration for RegNetY.
+
+    Args:
+        name: Name of the model eg. "RegNetY200MF"
+        flops: Flops of the model eg. "400MF"
+        depths: List of depths for every stage
+        widths: List of  widths (number of channels) after every stage
+        group_width: Integer denoting groups in every convolution layer
+        bottleneck_ratio: Integer specifying bottleneck ratio
+        SE_ratio: Float denoting squeeze and excite ratio
+        wa: Integer, slope used in linear parameterization
+        w0: Integer, inital value used in linear parameterization
+        wm: Float, quantization parameter
+    """
+
+    name: str
+    flops: str
+    num_classes: int
+    depths: List[int]
+    widths: List[int]
+    group_width: int
+    bottleneck_ratio: int
+    SE_ratio: float
+    wa: int
+    w0: int
+    wm: float
+
+
+
+def get_model_config(flops: str):
+    """
+    Getter function for configuration for a specific RegNetY model instance.
+    User must provide flops in string format. Example "200MF", "800MF" 
+    
+    Args: 
+        flops: String denoting flops of model (as per the paper)
+
+    Returns:
+        A RegNetYConfig dataclass instance with all attributes.
+    """
+    
+    if flops == "":
+        raise ValueError("Please enter `flops` argument.")
+    
+    
+
+    if flops.lower() not in ALLOWED_FLOPS:
+        raise ValueError("`flops` must be one of " + str(ALLOWED_FLOPS))
+    
+    if flops.lower() == '200mf':
+        return RegNetYConfig(
+            name = "RegNetY 200MF",
+            flops = "200MF",
+            num_classes = 1000,
+            depths = [1, 1, 4, 7],
+            widths = [24, 56, 152, 368],
+            group_width = 8,
+            bottleneck_ratio = 1,
+            SE_ratio = 0.25,
+            wa = 36,
+            w0 = 24,
+            wm = 2.5
+        )
+        
+    elif flops.lower() == '400mf':
+        return RegNetYConfig(
+            name = "RegNetY 400MF",
+            flops = "400MF",
+            num_classes = 1000,
+            depths = [1, 3, 3, 6],
+            widths = [48, 104, 208, 440],
```

Comment:
This relationship was derived from the explorations conducted in the paper. A linear quantized relationship if I am remembering correctly. Do we want to note this at the top? Just a single line would suffice.